students/becky_larson/lesson03/3.1slicingLab.py

```diff
@@ -1,77 +1,88 @@
 #!/usr/bin/env python """
 """Exchange first and last values in the list and return"""
-"""I learned that using the = sign, they refer to the same object, so I used copy.  Could also use list command"""
+"""learned that using = sign, they refer to same object, so used copy."""
+"""      Could also use list command"""
 
-"""tuple assert failing:  RETURNING:  (32, (54, 13, 12, 5, 32), 2). How to remove the parans"""
+"""tuple assert failing:    RETURNING:    (32, (54, 13, 12, 5, 32), 2). """
+"""      How to remove the parans"""
 """Is there a way to use same code for both string and tuple?"""
+
+
 def exchange_first_last(seq):
-  if(type(seq)) == str:
-    a_new_sequence = seq[len(seq)-1] + seq[1:(len(a_string)-1)] + seq[0]
-  else:
-    a_new_sequence = seq[len(seq)-1] ,seq[1:(len(a_string)-1)] , seq[0]
-  print("RETURNING: ", a_new_sequence)
-  return a_new_sequence
+    if(type(seq)) == str:
+        a_new_sequence = seq[len(seq)-1] + seq[1:(len(seq)-1)] + seq[0]
+    else:
+        a_new_sequence = []
+        a_new_sequence.append(seq[len(seq)-1])
+        a_new_sequence.extend(seq[1:(len(seq)-1)])
+        a_new_sequence.append(seq[0])
```

Comment:
your code should be the same for string, list or tuple - it's all using slicing.

the reason your code on line 13 is not working is because you're mixing slicing and indexing syntax.